### PR TITLE
add bash completion for puppet

### PR DESCRIPTION
**Pull request by saimonn**
puppet file stolen from debian bash-completion package.

### CODE REVIEW

**Comment by binford2k** [comment]
This completion script looks way out of date. What puppet version is Debian shipping now?

**Comment by saimonn** [comment]
The installable version of puppet in debian sid is currently 4.5.2-1, but this completion script comes from the bash-completion package, version 1:2.1-4.3

File: manifests/bash.pp [hunk 1]
```diff
@@ -31,6 +31,20 @@
...
   }
 
+  file { '/etc/bash_completion.d/puppet':
+    ensure  => file,
+    replace => $replace,
+    source  => 'puppet:///modules/userprefs/shell/completion/puppet',
+    require => Package['bash'],
+  }
+
+  file { ['puppetca', 'puppetd', 'puppetdoc', 'puppetmasterd', 'puppetqd', 'puppetrun']:
```
Comment (binford2k) [inline manifests/bash.pp:41]: We don't want any of these links. These commands don't exist anymore.

**Comment by binford2k** [comment]
Thanks for the PR and the suggestion. I'm going to rewrite the bash_completion script in a bit and then I'll commit it. I'll use some of our existing script (https://github.com/binford2k/bash_completion/blob/hard_code_opts/bash_completion.d/puppet) and some from this one, so this was really useful.

**Pull request merged**